msal/src/main/java/com/microsoft/identity/client/PublicClientApplication.java

```diff
@@ -1618,6 +1623,49 @@ public void onError(MsalException exception) {
         }
     }
 
+    public void deviceCodeFlow(@Nullable String[] scopes, @NonNull final DeviceCodeFlowCallback callback) {
+        // Create a DeviceCodeFlowCommandParameters object that takes in the desired scopes and the callback object
+        // Use CommandParametersAdapter
+        final DeviceCodeFlowCommandParameters commandParameters = CommandParametersAdapter
+                    .createDeviceCodeFlowCommandParameters(
+                            mPublicClientConfiguration,
+                            mPublicClientConfiguration.getOAuth2TokenCache(),
+                            scopes);
+
+        // Create a CommandCallback object from the DeviceCodeFlowCallback object
+        final DeviceCodeFlowCommandCallback dcfCallback = getDeviceCodeFlowCommandCallback(callback);
```

Comment:
nit: rename `dcfCallback` to `dcfCommandCallback` or `deviceCodeFlowCommandCallback`